src/lib/protocols/atg.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * atg.c
+ *
+ * Copyright (C) 2024 - ntop.org
+ *
+ * nDPI is free software: you can contribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_ATG
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+static void ndpi_int_atg_add_connection(struct ndpi_detection_module_struct
+						*ndpi_struct, struct ndpi_flow_struct *flow) {
+  ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_ATG,
+			     NDPI_PROTOCOL_UNKNOWN, NDPI_CONFIDENCE_DPI);
+}
+
+
+static void ndpi_search_atg(struct ndpi_detection_module_struct *ndpi_struct,
+				    struct ndpi_flow_struct *flow) {
+  struct ndpi_packet_struct *packet = &ndpi_struct->packet;
+  int i=1; 
+  
+  NDPI_LOG_DBG(ndpi_struct, "search for ATG\n");
+
+  if(packet->tcp != NULL) {
+  
+   if(packet->payload_packet_len >= 8) {
+      u_int16_t atg_port = ntohs(10001);
+
+    if((packet->tcp->source == atg_port) || (packet->tcp->dest == atg_port)) {
+	  if(packet->payload[0] == 0x01 ) {
+	  for(i=1;i<packet->payload_packet_len;i++){
+		if( packet->payload[i-1]==0x0d && packet->payload[i]==0x0a){
```

Comment:
You can replace it with just `if (memcmp(&packet->payload[packet->payload_packet_len-2], "\r\n", 2) == 0)`